### PR TITLE
Add generativeDirectAnswer service

### PR DESCRIPTION
**Pull request by anguyen-yext2**
J=CLIP-1226
TEST=auto,manual

ran and saw jest test pass.
tested the gda endpoint locally via core test site, saw appropriate api resp
onse.

### CODE REVIEW

**Review by EmilyZhang777 — COMMENTED** [review]
LGTM! But will defer to Clippy. Only have a nit

File: src/validation/ApiResponseValidator.ts [hunk 1]
```diff
@@ -33,6 +33,7 @@ export class ApiResponseValidator {
 
   private checkForApiErrors(apiResponse: ApiResponse): SearchError | undefined {
     if (apiResponse.meta?.errors?.length >= 1) {
+      apiResponse.meta.errors.map(e => console.log(e));
```
[the remark this answers is not in the document]
Comment (anguyen-yext2) [reply src/validation/ApiResponseValidator.ts]: oh oops. I forgot to remove it. Thanks for pointing it out!

File: src/models/generativedirectanswer/GenerativeDirectAnswerRequest.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ *
+ * @public
+ */
+export interface GenerativeDirectAnswerRequest extends SearchRequest {
+  /** The ID of the search request. */
+  searchId: string,
+  /** The text of the user-written query that prompted Search results. */
+  searchTerm: string,
+  /** The complete set of Search Results */
+  results: VerticalResults | Record<string, VerticalResults[]>
```
Comment (yen-tt) [inline src/models/generativedirectanswer/GenerativeDirectAnswerRequest.ts:15]: perhaps I missed something, but why is the type a map of key to results? why not just an array of VerticalResults?

Comment (anguyen-yext2) [reply src/models/generativedirectanswer/GenerativeDirectAnswerRequest.ts:15]: we defined results in [GDAApiRequestBody](https://gerrit.yext.com/plugins/gitiles/alpha/+/refs/heads/master/src/com/yext/search/generativeanswers/generative_answers.proto#18) to be of type struct, so we can't pass in array. `VerticalResults` is for a Vertical Search Result, `Record<string, VerticalResults[]>` is for a Universal Search Result. 
I think there is another way I could accomplish this: keep results to be of type `VerticalResults[]`, then in `GDAServiceImpl`, convert request.results to object before sending the api request. Something like this:
```
let results: Object = {}
if (request.results.length === 1) {
    results = request.results[1]
} else if (request.results.length > 1) {
    results = {verticals: request.results}
}
```
let me know if you prefer this way, then I can change and test it

Comment (yen-tt) [reply src/models/generativedirectanswer/GenerativeDirectAnswerRequest.ts:15]: yeah, I think that would work better so user of core don't have to transform the universal results themselves

File: src/infra/GenerativeDirectAnswerServiceImpl.ts [hunk 3]
```diff
@@ -51,10 +51,16 @@ export class GenerativeDirectAnswerServiceImpl implements GenerativeDirectAnswer
       visitorIdMethod: this.config.visitor?.idMethod,
       ...this.config?.additionalQueryParams
     };
+    let results = {};
```
Comment (yen-tt) [inline src/infra/GenerativeDirectAnswerServiceImpl.ts:54]: nit: add type

**Review by yen-tt — COMMENTED** [review]
are you going to publish a new version or wait for search's publish cycle? If it's the former, then you should bump the version in package.json

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@yext/search-core",
-  "version": "2.5.2",
+  "version": "2.6.0.beta",
```
Comment (yen-tt) [inline package.json:3]: super nit: can we follow the past version format from Search and use hyphen instead? ([example](https://www.npmjs.com/package/@yext/chat-ui-react?activeTab=versions)L 2.6.0-beta)

**Pull request merged**